### PR TITLE
chore(demo-playwright): implement addon-commerce tests

### PR DESCRIPTION
**Pull request by aotulana**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [X] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

Addresses #4912 

## What is the new behaviour?
This PR adds playwright e2e test for input card grouped and input expire, and removes the corresponding cypress tests

### CODE REVIEW

File: projects/demo-playwright/utils/page-objects/input-card-grouped.po.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    );
+
+    readonly disableFormControlToggle = this.page
+        .locator(`tr`)
+        .filter({
+            has: this.page.locator(`[automation-id="tui-documentation__property-name"]`, {
+                hasText: `disabled`,
+            }),
+        })
+        .locator(`[automation-id="tui-toggle__checkbox"]`);
```
Comment (nsbarsukov) [inline projects/demo-playwright/utils/page-objects/input-card-grouped.po.ts:19]: This method to disable component is just legacy way (when API pages did not support query param for `disable`).
Now we can just add it to url:
https://taiga-ui.dev/components/input-card-grouped/API?disabled=true

```ts
await tuiGoto(page, `components/input-card-grouped/API?disabled=true`);
```

____

Delete this property from `TuiInputCardGroupedPO` and update tests, please

File: projects/demo-playwright/utils/page-objects/input-card-grouped.po.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    getExampleCardGroupInput(example: Locator): Locator {
+        return example.locator(this.cardGroupInput);
+    }
+
+    getExampleCleanerIcon(example: Locator): Locator {
+        return example.locator(this.cleanerIcon);
+    }
+
+    getExampleCardExpiryInput(example: Locator): Locator {
```
Comment (nsbarsukov) [inline projects/demo-playwright/utils/page-objects/input-card-grouped.po.ts:31]: Page objects should know nothing about "external environment".

Drop all mentions about `example`

File: projects/demo-playwright/utils/page-objects/input-card-grouped.po.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    readonly disableFormControlToggle = this.page
+        .locator(`tr`)
+        .filter({
+            has: this.page.locator(`[automation-id="tui-documentation__property-name"]`, {
+                hasText: `disabled`,
+            }),
+        })
+        .locator(`[automation-id="tui-toggle__checkbox"]`);
+
+    constructor(private readonly page: Page) {}
```
Comment (nsbarsukov) [inline projects/demo-playwright/utils/page-objects/input-card-grouped.po.ts:21]: See other PO examples and make `TuiInputCardGroupedPO` to be similar to them

https://github.com/taiga-family/taiga-ui/blob/0e61adf8a57b87415879792cd856b65531c265e0/projects/demo-playwright/utils/page-objects/input-date-range.po.ts#L7

**Review by nsbarsukov — CHANGES_REQUESTED** [review]

File: projects/demo-playwright/utils/page-objects/input-card-grouped.po.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    readonly cleanerIcon = this.host.locator(
+        `[automation-id="tui-input-card-grouped__cleaner"]`,
+    );
+
+    readonly cardExpiryInput = this.host.locator(
+        `[automation-id="tui-input-card-grouped__expire"]`,
+    );
+
+    readonly cardCvcInput = this.host.locator(
```
Comment (nsbarsukov) [inline projects/demo-playwright/utils/page-objects/input-card-grouped.po.ts:16]: All other PO uses term "textfield".
Let's rename `input` and drop prefix `card`:

```suggestion
    readonly numberTextfield = this.host.locator(
        `[automation-id="tui-input-card-grouped__card"]`,
    );

    readonly cleanerIcon = this.host.locator(
        `[automation-id="tui-input-card-grouped__cleaner"]`,
    );

    readonly expiryTextfield = this.host.locator(
        `[automation-id="tui-input-card-grouped__expire"]`,
    );

    readonly cvcTextfield = this.host.locator(
```

File: projects/demo-playwright/utils/page-objects/input-card.po.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import {Locator} from '@playwright/test';
+
+export class TuiInputCardPO {
+    readonly cardExpiryInput = this.host
```
Comment (nsbarsukov) [inline projects/demo-playwright/utils/page-objects/input-card.po.ts:4]:
```suggestion
    readonly expiryTextfield = this.host
```

File: projects/demo-playwright/tests/addon-commerce/input-expire.spec.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    let cardExpiryInput: Locator;
+    let entryValue: string;
+    let expectedValue: string;
+
+    beforeEach(async ({page}) => {
+        await tuiGoto(page, `components/input-card/API`);
+
+        documentationPage = new TuiDocumentationPagePO(page);
+        example = documentationPage.apiPageExample;
+        inputCardPage = new TuiInputCardPO(example);
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/addon-commerce/input-expire.spec.ts:19]:
```suggestion
        inputCard = new TuiInputCardPO(example);
```

File: projects/demo-playwright/utils/page-objects/input-card-grouped.po.ts (hunk 4 above)
Comment (nsbarsukov) [reply projects/demo-playwright/utils/page-objects/input-card-grouped.po.ts:16]: Usage example:
```ts
const inputCard = new TuiInputCardPO(example);

await inputCard.expiryTextfield.pressSequentially('123');
await expect(inputCard.expiryTextfield).toHaveValue('123');
```

File: projects/demo-playwright/tests/addon-commerce/input-expire.spec.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    let example: Locator;
+    let cardExpiryInput: Locator;
+    let entryValue: string;
+    let expectedValue: string;
+
+    beforeEach(async ({page}) => {
+        await tuiGoto(page, `components/input-card/API`);
+
+        documentationPage = new TuiDocumentationPagePO(page);
+        example = documentationPage.apiPageExample;
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/addon-commerce/input-expire.spec.ts:18]:
```suggestion
       const {apiPageExample} = new TuiDocumentationPagePO(page);
```

File: projects/demo-playwright/tests/addon-commerce/input-expire.spec.ts [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+const {describe, beforeEach} = test;
+
+describe(`InputExpire`, () => {
+    let documentationPage: TuiDocumentationPagePO;
+    let inputCardPage: TuiInputCardPO;
+    let example: Locator;
+    let cardExpiryInput: Locator;
+    let entryValue: string;
+    let expectedValue: string;
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/addon-commerce/input-expire.spec.ts:12]: Why do we need to share these variables between tests ? 🤔

File: projects/demo-playwright/tests/addon-commerce/input-card-grouped.spec.ts [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+    TuiDocumentationPagePO,
+    tuiGoto,
+    TuiInputCardGroupedPO,
+} from '@demo-playwright/utils';
+import {expect, Locator, test} from '@playwright/test';
+
+const {describe, beforeEach, use} = test;
+
+describe(`InputCardGrouped`, () => {
+    use({viewport: {width: 1280, height: 800}});
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/addon-commerce/input-card-grouped.spec.ts:11]: Let use
```ts
use({viewport: {width: 700, height: 800}});
```

for `describe('API')`

___

And use

```ts
use({viewport: {width: 1280, height: 800}});
```

for `describe('Examples')`

File: projects/demo-playwright/tests/addon-commerce/input-expire.spec.ts (hunk 8 above)
Comment (nsbarsukov) [reply projects/demo-playwright/tests/addon-commerce/input-expire.spec.ts:12]: The only variable you actually use between all tests is `cardExpiryInput`.

File: projects/demo-playwright/tests/addon-commerce/input-card-grouped.spec.ts [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+        let cardCvcInput: Locator;
+
+        beforeEach(async ({page}) => {
+            await tuiGoto(page, `components/input-card-grouped`);
+
+            documentationPage = new TuiDocumentationPagePO(page);
+        });
+
+        test(`input card grouped with validation`, async () => {
+            example = documentationPage.getExample(`#validation`);
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/addon-commerce/input-card-grouped.spec.ts:71]: No need to "share" variable as you create this variable in every test.

```suggestion
            const example = documentationPage.getExample(`#validation`);
```

Comment (nsbarsukov) [reply projects/demo-playwright/tests/addon-commerce/input-card-grouped.spec.ts:71]: (the same for other variables)

File: projects/demo-playwright/tests/addon-commerce/input-card-grouped.spec.ts [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+            inputCardGroupedPage = new TuiInputCardGroupedPO(example);
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/addon-commerce/input-card-grouped.spec.ts:72]:
```suggestion
            inputCardGrouped = new TuiInputCardGroupedPO(example);
```

**Review by vladimirpotekhin — APPROVED** [review]

**Review by splincode — APPROVED** [review]

**Review by splincode — APPROVED** [review]

**Pull request merged**